astrosat_users/views/views_auth.py

```diff
@@ -140,18 +143,35 @@ class LoginView(RestAuthLoginView):
 
     permission_classes = [IsNotAuthenticated]
 
-    def get_response(self):
-        # note that this uses the KnoxTokenSerializer
-        # which has custom token validation
-        # which includes the LoginSerializer
-        # which adds extra astrosat_users validation
+    def get_success_response(self):
+        # this creates a response based on the KnoxTokenSerializer
+        # (and its nested LoginSerializer)
         serializer_class = self.get_response_serializer()
-
         data = {"user": self.user, "token": self.token}
         serializer = serializer_class(instance=data, context={"request": self.request})
-
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+    def get_error_response(self):
+        # this creates a response based on the state of the LoginSerializer
+        # (w/ some minimal user info as needed)
+        error_details = self.serializer.errors
+        if self.serializer.instance is not None:
+            user_serializer_lite = UserSerializerLite(
+                self.serializer.instance.get("user")
+            )
+            error_details["user"] = user_serializer_lite.data
+        return Response(error_details, status=status.HTTP_400_BAD_REQUEST)
+
+    def post(self, request, *args, **kwargs):
+        self.request = request
```

Comment:
This all looks find, I guess I'm a little confused as to the need to change from `local` to `instance` variables.

Reply:
I am overwriting the base class - the one from **dj_rest_auth** - `post` method here.  And this is what _that_ method did.  I am just ensuring that the instance variable is there in case any other base class methods need it.  (For what it's worth, saving a view's request as an instance variable is a standard thing that is done for all DRF views)